django/cantusdb_project/main_app/views/views.py

```diff
@@ -744,6 +744,32 @@ def json_node_export(request, id: int) -> JsonResponse:
     return HttpResponseNotFound()
 
 
+def notation_json_export(request, id: int) -> JsonResponse:
+    """
+    Return all fields of the notation with the specified ID
+    """
+
+    notation_qs: QuerySet[Notation] = Notation.objects.filter(id=id)
+    if len(notation_qs) == 0:
+        return HttpResponseNotFound()
+
+    values: dict = dict(*notation_qs.values())
+    return JsonResponse(values)
+
+
+def provenance_json_export(request, id: int) -> JsonResponse:
+    """
+    Return all fields of the provenance with the specified ID
+    """
+
+    provenance_qs: QuerySet[Provenance] = Provenance.objects.filter(id=id)
+    if len(provenance_qs) == 0:
+        return HttpResponseNotFound()
+
+    values: dict = dict(*provenance_qs.values())
```

Comment:
I feel like I’m missing something here. You’re doing a filter on an id, which likely means you want to do a get instead.

A QuerySet returns a list-like object, but you’re expanding it to a dict.

Couldn’t everything here be replaced with `get_or_404(id=id).values()`

Reply:
I followed the code already existing around https://github.com/DDMAL/CantusDB/blob/60606abb8a3f3a95837138fcb62c72c30dd098b3/django/cantusdb_project/main_app/views/views.py#L733 - While querysets can easily be converted into dictionaries via `.values()`, individual `Provenance` objects don't have a `.values()` method.

At the very least, I should have left a comment about why I was taking this approach. But even with a comment, this feels hacky - perhaps the correct approach here is to manually define a dictionary for each view.